flow360/component/results/case_results.py

```diff
@@ -462,26 +533,104 @@ class SurfaceForcesResultCSVModel(ResultCSVModel):
     remote_file_name: str = pd.Field(CaseDownloadable.SURFACE_FORCES.value, frozen=True)
 
 
-class ForceDistributionResultCSVModel(ResultCSVModel):
-    """ForceDistributionResultCSVModel"""
+class PerEntityResultCSVModel(ResultCSVModel):
+    _variables: List[str] = []
+    _x_columns: List[str] = []
+    _entities: List[str] = None
 
-    remote_file_name: str = pd.Field(CaseDownloadable.FORCE_DISTRIBUTION.value, frozen=True)
+    @property
+    def values(self):
+        """
+        Get the current data.
 
-    def wait(self, timeout_minutes=60):
-        """Wait until the Case finishes processing, refresh periodically"""
+        Returns
+        -------
+        dict
+            Dictionary containing the current data.
+        """
+        if self._values is None:
+            super().values
+            self._sum()
+        return super().values
 
-        start_time = time.time()
-        while time.time() - start_time < timeout_minutes * 60:
-            try:
-                self.load_from_remote(log_error=False)
-                return None
-            except CloudFileNotFoundError:
-                pass
-            time.sleep(2)
+    @property
+    def entities(self):
+        """
+        Returns list of entities (boundary names) available for this result
+        """
+        if self._entities is None:
+            pattern = re.compile(rf"(.*)_({'|'.join(self._variables)})$")
+            prefixes = {
+                match.group(1) for col in self.as_dict().keys() if (match := pattern.match(col))
+            }
+            self._entities = sorted(prefixes)
+        return self._entities
+
+    def filter(self, include: list = None, exclude: list = None):
```

Comment:
As a side note. What do you think of the CSV file format that we use for now? The CSV seems does not have enough capability to achieve what we want.

For example here properly associating entities with a field ( we have to do hard coded naming patter like entity_name+field_name for now). Does alternative have better data structure to store metadata? Like HDF5?

Reply:
yes, I think HDF5 would be better idea. User can convert this data to CSV with column names he prefers to use. I think HDF5 + pandas DataFrame + saving as CSV would be better and more scalable solution. And user interface will be not affected by internal data storage conventions.